pattern/Conjunction.java

```diff
@@ -61,26 +62,27 @@
 import static com.vaticle.typeql.lang.common.TypeQLToken.Char.NEW_LINE;
 import static com.vaticle.typeql.lang.common.TypeQLToken.Char.SEMICOLON;
 import static com.vaticle.typeql.lang.common.TypeQLToken.Char.SPACE;
+import static java.util.Collections.unmodifiableList;
 import static java.util.Collections.unmodifiableMap;
 import static java.util.Collections.unmodifiableSet;
-import static java.util.stream.Collectors.toSet;
+import static java.util.stream.Collectors.toList;
 
 public class Conjunction implements Pattern, Cloneable {
 
     private static final String TRACE_PREFIX = "conjunction.";
     private final Map<Identifier.Variable, Variable> variableMap;
     private final Set<Variable> variableSet;
-    private final Set<Negation> negations;
+    private final List<Negation> negations;
```

Comment:
key change: store negations in a `List` instead of a `Set`, which can use negations due to ill-defined equality functions.